src/ucp/rndv/rndv_ppln.c

```diff
@@ -156,15 +156,15 @@ ucp_proto_rndv_ppln_frag_complete(ucp_request_t *freq, int send_ack,
         req->send.rndv.ppln.ack_data_size += freq->send.state.dt_iter.length;
     }
 
-    if (!ucp_proto_rndv_frag_complete(req, freq, title)) {
+    if (!ucp_proto_rndv_frag_complete(req, freq, title) && !abort) {
```

Comment:
is it possible we would complete more fragments of the same top-level request, but then their "super" would point to already released request?

Reply:
I don't see the case for `rndv/rtr/mtype` but it's possible for other ppln or multi protocols like `rndv/get/mtype`, for example, which does not have `abort` implementation yet. In such case, I think `abort` flag should be set by completion status to handle outstanding fragments.

Reply:
I don't think it will help if we complete the request immediately; additional fragments may arrive and use an already-released request.
you are right it cannot happen for RTR/mtype because we search for request by its ID and now we'd remove it from the hash.

Reply:
So, do you agree with this implementation of RTR or request some change?

Reply:
yes, current rtr abort is ok because of request id hash.